submissions/krylenger/memory-pair-game/app.js

```diff
@@ -0,0 +1,244 @@
+    main.appendChild(mainGridContainer);
+}
+
+const resetGame = () => {
+    recreateMainGridContainer();
+    loadCards(cards);
+    guessedCards = [];
+    playedSongs = [];
+    handleClick();
+}
+
+const handlePlayingCards = (cardStatus) => {
+    const playingNow = document.querySelectorAll(".playingNow");
+    playingNow.forEach((card) => {
+        card.classList.remove("playingNow");
+        if (cardStatus === 'guessed') {
+            setTimeout(() => {
+                card.parentNode.parentNode.classList.add('card--background-green');
+            }, 300)
+        }; 
+        setTimeout(() => {
+            card.parentNode.classList.toggle(cardStatus);
+        }, 1000)
+    });
+}
+
+
+const handleGameOver = (guessedCards) => {
+    if (guessedCards === 10) {
+        setTimeout(() => {
+            alert(`Congratulations! Let's play once more!`);
+            resetGame(guessedCards);
+        }, 1000)
+    }
+}
+
+const stopPreviousSong = (playedSongs) => {
+    let previousSong = playedSongs[playedSongs.length - 1];
+    previousSong.pause();
+    previousSong.currentTime = 0;
+}
+
+const handleMusicCard = (target) => {
+    if (target.parentNode.parentNode.classList.contains('musicCard')) {
+        const targetKeyNumber = target.classList.value.match(/\d/)[0];
+        const targetObj = cards.find((card) => card.musicCard && card.key === Number.parseInt(targetKeyNumber, 10));
+        if (playedSongs.length) {
+            stopPreviousSong(playedSongs);
+        }
+        playedSongs.push(targetObj.song);
+        targetObj.song.play();
+        
+    };
+}
+
+const flipTwoCards = () => {
+    let clickCounter = 0;
+    let guessedCards = 0;
+    let playingCards = [];
+    mainGridContainer.addEventListener('click', ({target}) => {
+        if (
+          target.classList.contains("card__flipper-front") &&
+          !target.parentNode.classList.contains("guessed") &&
+          playingCards.length < 2
+        ) {
+          target.parentNode.classList.toggle("card__flipper--flip");
+          target.classList.add("playingNow");
+          handleMusicCard(target);
+
+          clickCounter += 1;
+          playingCards.push(target.classList.value);
+          if (clickCounter === 2 && playingCards[0] === playingCards[1]) {
+            handlePlayingCards('guessed');
+            modal.classList.add('modal--visible');
+            setTimeout(() => {
+                modal.classList.remove('modal--visible');
+            }, 1200)
+            clickCounter = 0;
+            playingCards = [];
+            guessedCards += 2;
+          } else if (clickCounter === 2 && playingCards[0] !== playingCards[1]) {
+            handlePlayingCards('card__flipper--flip');
+            setTimeout(() => {
+                clickCounter = 0;
+                playingCards = [];
+            }, 1000)
+          }
+        }
+        handleGameOver(guessedCards);
+    })
+}
+
+const handleClick = () => {
```

Comment:
Do you really need to wrap the `flipTwoCards` call into another function?

Reply:
I thought it's gonna be better in 'reading' code matter but now I see that this function is really redundant.
✅ Removed